src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/SPARQL2GraphQLTranslatorImpl.java

```diff
@@ -0,0 +1,454 @@
+                    final String sgpType = object.getURI().substring(splitIndex);
+                    return sgpType;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @return a map consisting of what can be used as arguments from the given @param sgp
+     * The predicate from a TP needs to be a property URI and the object needs to be a literal
+     */
+    protected final Map<String,LiteralLabel> getSgpArguments(final Set<TP> sgp, final GraphQL2RDFConfiguration config){
+        final Map<String,LiteralLabel> args = new HashMap<>();
+        for(final TP t : sgp){
+            final Node predicate = t.getTriplePattern().asJenaTriple().getPredicate();
+            final Node object = t.getTriplePattern().asJenaTriple().getObject();
+
+            if(predicate.isURI() && predicate.getURI().startsWith(config.getPropertyPrefix())){
+                if(object.isLiteral() && object.getLiteral().isWellFormed()){
+                    // Remove prefix and suffix before adding to map
+                    final String s = config.removePropertyPrefix(predicate.toString());
+                    final String argName = config.removePropertySuffix(s);
+                    args.put(argName,object.getLiteral());
+                }
+            }
+        }
+        return args;
+    }
+
+    /**
+     * Checks whether a set of SGP arguments, @param sgpArguments , meets the necessery arguments for
+     * an endpoint @param entrypointArguments . If @param checkAll is true then all arguments from the entrypoint
+     * must exist in the SGP. Otherwise atleast one matching argument is enough as long as @param sgpArguments
+     * isn't an empty set.
+     */
+    protected boolean hasNecessaryArguments(final Set<String> sgpArguments, final Set<String> entrypointArguments, 
+            final boolean checkAll){
+        
+        if(sgpArguments.isEmpty()){
+            return false;
+        }
+
+        if(checkAll){
+            return sgpArguments.containsAll(entrypointArguments);
+        }
+        else{
+            for(final String argName : sgpArguments){
+                if(entrypointArguments.contains(argName)){
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Helper function used to convert a LiteralLabel value to a valid jsonvalue
+     */
+    protected JsonValue literalToJsonValue(final LiteralLabel literal){
+        final Class<?> type = literal.getValue().getClass();
+
+        if(type.equals(int.class)){
+            return JsonNumber.value((int) literal.getValue());
+        }
+        else if(type.equals(boolean.class)){
+            return new JsonBoolean((boolean) literal.getValue());
+        }
+        else if(type.equals(double.class) || type.equals(float.class)){
+            return JsonNumber.value((double) literal.getValue());
+        }
```

Comment:
My IDE tells me that these conditions are always `false`. Probably because `int.class` and such return something with a different type than `Class<?>`. Please reconsider the utility of this method.